runelite-client/src/main/java/net/runelite/client/plugins/chatfilter/ChatFilterPlugin.java

```diff
@@ -131,7 +133,15 @@ public void onScriptCallbackEvent(ScriptCallbackEvent event)
 		String message = stringStack[stringStackSize - 1];
 		String censoredMessage = censorMessage(message);
 
-		if (censoredMessage == null)
+		if (chatMessageType == LOGINLOGOUTNOTIFICATION)
```

Comment:
This whole if-else-if-else block can be simplified to:

```java
if ((chatMessageType == LOGINLOGOUTNOTIFICATION && config.filterLogin()
	&& (messageNode.getValue().endsWith("has logged in.") || messageNode.getValue().endsWith("has logged out.")))
	|| censoredMessage == null)
{
	// Block the message
	intStack[intStackSize - 3] = 0;
}
else
{
	// Replace the message
	stringStack[stringStackSize - 1] = censoredMessage;
}
```

Reply:
I like current format more because its actually readable :d

Reply:
That's what I think as well too